sdk/assemblyscript/src/assembly/agent.ts

```diff
@@ -271,3 +288,27 @@ export class AgentInfo {
     this.status = status;
   }
 }
+
+/**
+ * Base class for agent events.
+ * Custom agent events should extend this class.
+ */
+@json
+export class AgentEvent {
+  /**
+   * The name of the event.
+   */
+  @omit
+  readonly eventName: string;
+
+  /**
+   * Creates a new agent event.
+   * @param eventName The name of the event.
+   */
+  constructor(eventName: string) {
```

Comment:
I wonder if the type of the payload and eventName should be colocated somehow.

Reply:
I thought about that, but it seemed easier to be verbose.  Ultimately it only matters how they are serialized.